source/operators/captions/webcaptions.py

```diff
@@ -0,0 +1,1016 @@
+            self.marker = "<span>"
+            self.contentType = "text/html"
+            self.existing_subtitles = False
+            if "SourceLanguageCode" in self.operator_object.configuration:
+                self.source_language_code = self.operator_object.configuration["SourceLanguageCode"]
+                self.operator_name_with_lang = self.operator_object.name+"_"+self.source_language_code
+            if "TargetLanguageCode" in self.operator_object.configuration:
+                self.target_language_code = self.operator_object.configuration["TargetLanguageCode"]
+            if "ExistingSubtitlesObject" in self.operator_object.configuration:
+                self.existing_subtitles_object = self.operator_object.configuration["ExistingSubtitlesObject"]
+                self.existing_subtitles = True
+        except KeyError as e:
+            self.operator_object.update_workflow_status("Error")
+            self.operator_object.add_workflow_metadata(WebCaptionsError="No valid inputs {e}".format(e=e))
+            raise MasExecutionError(operator_object.return_output_object())
+
+    def WebCaptionsOperatorName(self, language_code=None, source=""):
+
+        # Shouldn't assume WebCaptions operator is WebCaptions, maybe pass it in the configuration?
+        print("WebCaptionsOperatorName {}, {}".format(language_code, source))
+
+        operator_name = "WebCaptions"+source
+
+        if language_code != None:
+            return operator_name+"_"+language_code
+
+        try:
+            name = operator_name+"_"+self.source_language_code
+        except KeyError:
+            self.operator_object.update_workflow_status("Error")
+            self.operator_object.add_workflow_metadata(WebCaptionsError="Missing language code for WebCaptions {e}".format(e=e))
+            raise MasExecutionError(self.operator_object.return_output_object())
+
+        print("WebCaptionsOperatorName() Name {}".format(name))
+        return name
+
+    def CaptionsOperatorName(self, language_code=None):
+
+        # Shouldn't assume WebCaptions operator is WebCaptions, maybe pass it in the configuration?
+        operator_name = "Captions"
+
+        if language_code != None:
+            return operator_name+"_"+language_code
+
+        try:
+            name = operator_name+"_"+self.source_language_code
+        except KeyError:
+            self.operator_object.update_workflow_status("Error")
+            self.operator_object.add_workflow_metadata(WebCaptionsError="Missing language code for WebCaptions {e}".format(e=e))
+            raise MasExecutionError(self.operator_object.return_output_object())
+
+        print("CaptionsOperatorName() Name {}".format(name))
+        return name
+
+
+
+    def GetTranscript(self):
+
+        transcript = []
+
+        response = dataplane.retrieve_asset_metadata(self.asset_id, operator_name=self.transcribe_operator_name)
+        transcript.append(response["results"])
+
+        while "cursor" in response:
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

To create a `list`, try to use `list` comprehension instead of a loop. List comprehension is the preferred way to make a list using Python, and it&#39;s simpler and easier to understand than using a loop.

[Learn more](https://docs.python.org/3/tutorial/datastructures.html#list-comprehensions)

Reply:
We are not going to address this CodeGuru finding because we want to mirror the official AWS code sample for sigv4 signing: https://docs.aws.amazon.com/code-samples/latest/catalog/python-signv4-v4-signing-get-post.py.html.